rank_pairing/rank_pairing_heap.go

```diff
@@ -0,0 +1,334 @@
+package rank_paring
+
+import (
+	"fmt"
+
+	heap "github.com/theodesp/go-heaps"
+)
+
+type node struct {
+	Val                heap.Item
+	Left, Next, Parent *node
+	Rank               int
+}
+
+// RPHeap is an implementation of a Rank Pairing Heap.
+// The zero value for RPHeap Root is an empty Heap.
+type RPHeap struct {
+	head *node
+	size int
```

Comment:
We don't really use size in all the heaps so far

Reply:
size is used in DeleteMin()